File: YogaKit/YogaKitSample/YogaKitSample/ViewController.m
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "ViewController.h"

#import <YogaKit/UIView+Yoga.h>

@implementation ViewController

- (void)viewDidLoad
{
    UIView *root = self.view;
    root.backgroundColor = [UIColor redColor];
    root.yoga.isEnabled = YES;
    root.yoga.width = YGPointValue(self.view.bounds.size.width);
    root.yoga.height = YGPointValue(self.view.bounds.size.height);
    root.yoga.alignItems = YGAlignCenter;
    root.yoga.justifyContent = YGJustifyCenter;

    UIView *child1 = [UIView new];
    child1.backgroundColor = [UIColor blueColor];
    child1.yoga.isEnabled = YES;
    child1.yoga.width = YGPointValue(100);
    child1.yoga.height = YGPointValue(100);

    UIView *child2 = [UIView new];
    child2.backgroundColor = [UIColor greenColor];
    child2.frame = (CGRect) {
        .size = {
            .width = 200,
            .height = 100,
        }
    };

    UIView *child3 = [UIView new];
    child3.backgroundColor = [UIColor yellowColor];
    child3.frame = (CGRect) {
        .size = {
            .width = 100,
            .height = 100,
        }
    };

    [child2 addSubview:child3];
    [root addSubview:child1];
    [root addSubview:child2];
    [root.yoga applyLayoutPreservingOrigin:NO];
}


@end

File: csharp/iOS/Facebook.YogaKit.iOS.Sample/ViewController.cs
using System;
using CoreGraphics;
using Facebook.Yoga;
using UIKit;

namespace Facebook.YogaKit.iOS.Sample
{
	public partial class ViewController : UIViewController
	{
		protected ViewController(IntPtr handle) : base(handle)
		{
			// Note: this .ctor should not contain any initialization logic.
		}

		public override void ViewDidLoad()
		{
			base.ViewDidLoad();
			CreateViewHierarchy(View, View.Bounds.Size.Width, View.Bounds.Size.Height);
		}

		static void CreateViewHierarchy(UIView root, nfloat width, nfloat height)
		{
			root.BackgroundColor = UIColor.Red;
			root.Yoga().IsEnabled = true;

			root.Yoga().Width = (float)width;
			root.Yoga().Height = (float)height;
			root.Yoga().AlignItems = YogaAlign.Center;
			root.Yoga().JustifyContent = YogaJustify.Center;

			var child1 = new UIView { BackgroundColor = UIColor.Blue };
			child1.Yoga().IsEnabled = true;
			child1.Yoga().Width = 100;
			child1.Yoga().Height = 100;

			var child2 = new UIView
			{
				BackgroundColor = UIColor.Green,
				Frame = new CGRect { Size = new CGSize(200, 100) }
			};

			var child3 = new UIView
			{
				BackgroundColor = UIColor.Yellow,
				Frame = new CGRect { Size = new CGSize(100, 100) }
			};

			child2.AddSubview(child3);
			root.AddSubview(child1);
			root.AddSubview(child2);
			root.Yoga().ApplyLayout();
		}
	}
}
